server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1834,6 +1836,17 @@ public void syncDatastoreClusterStoragePool(long datastoreClusterPoolId, List<Mo
         handleRemoveChildStoragePoolFromDatastoreCluster(childDatastoreUUIDs);
     }
 
+    private StoragePoolVO getExistingPoolByUuid(String uuid){
+        if(!uuid.contains("-")){
```

Comment:
I don't see the benefit. When I change that, then I need additionally a else statement or two return statements.

Reply:
My suggestion is to add two return statements. Inverting the `if` statement will reduce indentation and increase readability. Also, it will prevent this code to turn into a `hadouken code`.

Reply:
With your suggestion keeps the if statement and the indentation. I'm not sure what do you exactly mean and we talk about one if statement :D.

Reply:
What I meant:

```java
private StoragePoolVO getExistingPoolByUuid(String uuid){
    if (uuid.contains("-")) {
        return _storagePoolDao.findByUuid(uuid);
    }
    
    UUID poolUuid = new UUID(
        new BigInteger(uuid.substring(0, 16), 16).longValue(),
        new BigInteger(uuid.substring(16), 16).longValue()
    );
    uuid = poolUuid.toString();
    return _storagePoolDao.findByUuid(uuid);
}
``` 

or

```java
private StoragePoolVO getExistingPoolByUuid(String uuid){
    if (uuid.contains("-")) {
        return _storagePoolDao.findByUuid(uuid);
    }
    
    uuid = new UUID(
        new BigInteger(uuid.substring(0, 16), 16).longValue(),
        new BigInteger(uuid.substring(16), 16).longValue()
    ).toString();

    return _storagePoolDao.findByUuid(uuid);
}
```

Reply:
As I thought. But why should this code be better?

Reply:
Inverting the `if` statement will reduce indentation and increase readability. Also, it will prevent this code to turn into a `hadouken code`. 

It's not obligatory, it's just a suggestion.

Reply:
Hi @GutoVeronezi, thanks for your suggestion, but in this case I'll keep the current implementation.